x-pack/plugins/upgrade_assistant/server/lib/reindexing/index_settings.test.ts

```diff
@@ -177,4 +179,24 @@ describe('getReindexWarnings', () => {
       })
     ).toEqual([]);
   });
+
+  if (mockKibanaSemverVersion.major === 7) {
```

Comment:
Not sure how the test harness is being set up, but it could be a better pattern to create a nested `describe` blocks for the 7 cases and 8 cases with their own test setups. WDYT?